public/app/features/dashboard/components/GenAI/GenAIButton.tsx

```diff
@@ -24,41 +26,40 @@ export const GenAIButton = ({
   temperature = 1,
 }: GenAIButtonProps) => {
   const styles = useStyles2(getStyles);
-  const [enabled, setEnabled] = useState(true);
-  const [loading, setLoading] = useState(false);
 
-  const replyHandler = (response: string, isDone: boolean) => {
-    setLoading(!isDone);
-    onReply(response, isDone);
-  };
+  // TODO: Implement error handling (use error object from hook)
+  const { setMessages, reply, started, loading, value, error, finished } = useOpenaiStreamHook(
+    OPEN_AI_MODEL,
+    temperature
+  );
 
   const onGenerate = (e: React.MouseEvent<HTMLButtonElement>) => {
     onClick?.(e);
-    setLoading(true);
-    generateTextWithLLM(messages, replyHandler, temperature);
+    setMessages(messages);
   };
 
-  useEffect(() => {
-    isLLMPluginEnabled()
-      .then(setEnabled)
-      .catch(() => setEnabled(false));
-  }, []);
+  // Todo: Consider other options for `"` sanitation
+  if (started) {
+    onReply(reply.replace(/"/g, ''));
```

Comment:
This is going to remove any double quotes. Instead, we should just remove the one around the message, shouldn't we?

Reply:
The message has double quotes on it? When running the hook in llmexamples it's just a string, it won't have surrounding double quotes unless GPT adds them, which would be odd behavior.

Reply:
(My current feeling is we have enough merge conflicts already and we should set this issue aside as out of scope for this PR)

Reply:
I added this as GPT was returning results consistently surrounded by "" - this could also be addressed prompt engineering 🤷‍♂️

Reply:
We can keep it to make sure that the use case is covered, but we should just remove the quotes at the beginning and at the end of the string.

```js
response.replace(/^"|"$/g, '')
```